### DISCUSSION TITLE
BUG: Options for choosing "Categories" when saving Custom Scenario?

### DISCUSSION THREAD

**Discussion by sevvyb** [OpenRCT2/OpenRCT2#20932] [Feature requests]
Forgive me as this might seem confusing to explain...

I want to edit RCT1 scenarios with different objectives & invention list (which is easy to do) however, when I save Custom Scenarios it gives me the option of choosing a category, yet the only options available are "Beginner, Challenging, or Expert" (which are Sub-categories) and no matter what it places my Custom Scenario into the "Other Parks" category with no option to change it. How do I re-organize things??

I have downloaded many Custom Scenarios, usually they go into the "Other Parks" category, but sometimes Custom Scenarios show up in other categories too. I downloaded some called "RCT1 remakes for RCT2" which are all .SC6 format and they show up in the RCT1 category alongside the original RCT1 scenarios (so now I have duplicates, since original RCT1 scenarios use different format)

**how do I save my personal custom scenarios so they remain in the RCT1 category? It seems like this should be possible**

_Another example/issue_ - Fred104's scenario downloads use subfolder/subcategories ([see here](https://forums.openrct2.org/topic/5391-fred-104s-rct2-custom-scenarios-now-with-340-scenarios/))

however, Im using [UCES](https://uces.rctspace.com/) which seems to cause issues with subcategories, because UCES prevents the "Other Parks" category from using customized subfolders, as it seems UCES forces all categories to use "Beginner, Challenging, and Expert" subcategories... I believe thats a separate/different issue from my original question, but it also sort of proves that it must be possible to change/fix this somehow

Lastly - really wish I could re-organize my entire list of scenarios & categories, because I have wayyyy too many downloaded custom scenarios and I dont want them all together in the same category. I really think we need at least one additional Customizable Category to separate custom scenarios from downloads... and do we really need BYO parks separated from "Real" Parks??

Thanks :)

**Comment by MASHzj** [comment]
The difficulties are subcategories if you enable the option to sort the scenarios by origin game or expansion instead of by difficulty. Originally the difficulties are the main categories like in vanilla RCT2. That's why you can only choose the difficulty of your custom scenarios. I think when sorted by origin game it's hard coded which scenarios go in which category. Thus I don't think there is currently a way to move scenarios to another category.

**Reply by sevvyb** [reply]
> difficulties are subcategories if you enable the option to sort the scenarios by origin

Yes I prefer sorting by game origin/expansion

> you can only choose the difficulty of your custom scenarios

thats what we need!! the ability to set origin/expansion in the scenario editor

> I don't think there is currently a way to move scenarios to another category

My reason for believing this must be possible is because, I downloaded some Custom RCT2 Scenarios in .SC6 format which are placed in the RCT1 category... and what about UCES which has its own unique category?? thats 2 examples of it being possible!!

https://www.reddit.com/r/rct/comments/1h93gx/more_exact_recreations_of_rct1_scenarios_for_rct2/

many of ^ these are placed in the RCT1 category and some of them are placed under "Extras" for some reason, yet they do not overwrite or replace my original RCT1 scenarios, meaning I can have 2 versions of Forest Frontiers in the RCT1 category while leaving the originals untouched, since my original RCT install files are a different format

how were those RCT1 custom scenarios made?? I want to make my own versions like those posted above (but with my own objectives & invention list) if people were able to do it before with RCT1 and UCES, then it must still be possible somehow 

Personally I'd be happy just to pick the origin/expansion category for my custom scenarios, but it would be really amazing if we could create new categories too. Maybe we could have RCT1 originals, RCT Classic, and a set of remade RCT1 scenarios for OpenRCT2!!

**Reply by MASHzj** [reply]
I absolutely agree with you that having tabs by source game is much better than by difficulty, especially if you have expansions and/or RCT1 installed. I'm just clarifying that OpenRCT2 provides this option from the default by difficulty, but the logical next steps like choosing your tab in a custom scenario, and making your own tabs haven't been implemented so far. There has to be someone who is willing and able to implement it.

About the RCT1 recreations in RCT2: I guess the developers have programmed it that way that recreations would go in the same tab as the original scenarios. If some go in other tabs, it might be a bug then. I don't know exactly how the game recognizes a recreation scenario (park name, file name ... ?) to put into the correct category. So maybe there is a way to put your own variations of original scenarios under the same tab, I don't know. Maybe you can ask this on discord, where there is a higher chance that someone who knows more about this will answer.
Same for UCES, I guess that tab is just hardcoded to occur if you download UCES scenarios because it's a common custom scenario pack.

**Comment by sevvyb** [comment]
> choosing your tab in a custom scenario, and making your own tabs haven't been implemented so far

Yea obviously, thats what Im asking for... isnt this the "requests" section?? or am I in the wrong subforum?

> you can ask this on discord, where there is a higher chance that someone who knows more about this will answer.

Discord is annoying cuz I have to wait for somebody to talk back and forth while Im at my PC... I cant just post a question and leave, then expect to find a conversation when I come back (like Ive done here with this thread) its also rather complicated to explain in a few short lines

> There has to be someone who is willing and able to implement it.

Yea I agree (as Ive said) I believe there must be a way, I mean its obviously coded in the game somehow, its just a matter of finding somebody with the coding skills who is willing to figure out where its coded and how to edit it (isnt that what OpenRCT is all about??)

> maybe there is a way to put your own variations of original scenarios under the same tab

I already said that it is in fact possible, if u download the RCT1 remakes that I posted above, u can have 2 different sets of RCT1 scenarios in the RCT1 category, the RCT1 remake files are in SC6 format and will appear in the RCT1 Category

HOWEVER if I create my OWN custom scenario, it goes into "Other Parks" because I dont know the formatting rules

>  tab is just hardcoded to occur if you download UCES scenarios

I also posted on Reddit and got an answer:

_______
From Reddit user: "For the scenario to count as being one of those main scenarios, it needs to be named the same (i.e. sc0 and it cannot have the .park extension.) v0.3.5.1 is the last version before the new file format. Also, you can have the new scenarios just exist wherever in your data folder, it doesn't need to be within the RCT1 files.

So simply put, the only reason your new files are being put in the "other parks" section is because of the file format. This unfortunately means new features implemented with the new format cannot be saved to these scenarios, however. I would personally love to make more tabs or some kind of tree view, as has been discussed by the devs, to allow better organization of scenario packs. This would even allow for remastered versions of scenarios to exist with the new format and be organized like they are in vanilla."
_______

Im confused bcuz the RCT1 remakes are in SC6 format, but either way, this is clearly VERY much a possibility... Im not trying to say its an easy thing to do, Im simply saying it is "possible" and those who say it is not possible seem wrong or confused... According to the reply I received on Reddit, it has to do with the save file format and apparently some of the devs have already looked into it... So my hope is that maybe if enough people ask for it, then maybe the devs will look further into it. Thats all

SInce I just discovered the above details about SC0 formatting, I will try testing that out tomorrow when I have time and Ill update this post if I learn anything new :)
